STN-116 : Search Bar Polish

heatherdesigns opened the pull request:
# [STN-116](https://sparkbox.atlassian.net/browse/STN-116) READY FOR REVIEW

## Summary
Ensure the search bar looks and works as desired in the colorful theme.

Designs: 
[Figma desktop](https://www.figma.com/file/toNZnFLadtpNpG5dPzNGWz/⚙%EF%B8%8F-Colorful-Theme---Working?node-id=3603%3A282)
[Figma mobile](https://www.figma.com/file/toNZnFLadtpNpG5dPzNGWz/⚙%EF%B8%8F-Colorful-Theme-Working?node-id=3547%3A832)

## Need Review By (Date)
asap

## Urgency
high 🔥

## Steps to Test
1. In the CLI run `npm run test` to verify all tests still pass.
2. Go to http://economics.suhumsci.loc and verify the search input chisel meets the following acceptance criteria:
    - [x] Ensure that the search bar looks and works as expected after pairing with a designer and getting approval from the H&S team. I paired with Merani on design.
    - [x] Ensure that things work across all supported browsers.
    - [x] Ensure that things meet our accessibility guidelines

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
feat: polish search bar styles

JIRA STN-116 https://sparkbox.atlassian.net/browse/STN-116

Commit:
feat: edit styles so that the search bar remains visible in the masthead at all times

Commit:
feat: implement design feedback for search input

bryanbraun commented on docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:
```diff
@@ -12,35 +11,39 @@
 
     background-color: transparent;
-    background-image: svg('<svg viewBox="0 0 24 25"><path fill-rule="evenodd" clip-rule="evenodd" d="M21.4737 10.2016C21.4737 14.0385 18.3632 17.149 14.5263 17.149C10.6894 17.149 7.57895 14.0385 7.57895 10.2016C7.57895 6.36466 10.6894 3.25422 14.5263 3.25422C18.3632 3.25422 21.4737 6.36466 21.4737 10.2016ZM24 10.2016C24 15.4338 19.7585 19.6753 14.5263 19.6753C12.5029 19.6753 10.6277 19.0409 9.088 ...
+    background-image: svg('<svg viewBox="0 0 25 26"><path fill-rule="evenodd" clip-rule="evenodd" d="M21.4737 10.2016C21.4737 14.0385 18.3632 17.149 14.5263 17.149C10.6894 17.149 7.57895 14.0385 7.57895 10.2016C7.57895 6.36466 10.6894 3.25422 14.5263 3.25422C18.3632 3.25422 21.4737 6.36466 21.4737 10.2016ZM24 10.2016C24 15.4338 19.7585 19.6753 14.5263 19.6753C12.5029 19.6753 10.6277 19.0409 9.088 ...
```
Great use of the SCSS variables here!

bryanbraun commented on docroot/themes/humsci/humsci_basic/templates/page.html.twig:
```diff
@@ -3,13 +3,13 @@
 
     {% if page.search %}
-    <div class="su-site-search">
+    <div class="su-site-search" role="search">
```
Great addition!

silverli replied:
👍

Commit:
refactor: center nav toggle across all screens once the nav has collapsed

- remove absolute positioning from the search input field and add it to the menu toggle

kaseybon commented on docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:
```diff
@@ -12,35 +11,39 @@
-    right: 1.2rem;
-    width: 24px;
-    height: 25px;
+    right: 14px;
```
We should use our `hb-calculate-rems()` function here.

kaseybon reviewed (changes requested):
The search button alignment is off when logged out of Drupal and we've also lost the focus and hover states for the search button.

<img width="417" alt="Screen Shot 2020-02-26 at 4 06 03 PM" src="https://user-images.githubusercontent.com/2486846/75387881-fe78b280-58b1-11ea-995a-140ee7908b0d.png">

---

There are also some issues with the mobile nav icon not lining up and the mobile nav overlaying the search input. I don't think it's within the scope of this card but can you please make a note in STN-210 that these will need to be resolved?

edit: this is affecting dropdowns in desktop views too.

Commit:
feat: implement review feedback

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr499-nuinpeaifcojjeikvruwxzkyejynzz52.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e58067f8c1e65164ef749e8

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss:
```diff
@@ -1,3 +1,46 @@
+    padding-top: hb-calculate-rems(12px);
+    padding-bottom: hb-calculate-rems(28px);
+    max-width: hb-calculate-rems(265px);
+    z-index: $hb-z-index-2;
```
you could  create a named variable for the z index like you did on line 3 above. for instance, `search-bar-z-index` would be good!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss:
```diff
@@ -1,3 +1,46 @@
+    display: block;
+    padding-top: hb-calculate-rems(12px);
+    padding-bottom: hb-calculate-rems(28px);
+    max-width: hb-calculate-rems(265px);
```
let's put this in settigs/variables.general.scss and make a named variable for 256px

heatherdesigns replied:
I placed it in `_variables/forms` instead. Let me know if that's ok.

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss:
```diff
@@ -1,3 +1,46 @@
+      position: absolute;
+      right: 0;
+      bottom: 80%;
+      transform: translate(0, -80%); // Center the toggle on nav styles implementation. It is not centered while using Decanter's default nav styles because they overlay ontop of the lockup which makes the content inaccessible.
```
great comment!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_masthead.scss:
```diff
@@ -1,3 +1,46 @@
+      }
+    }
+
+    // The positioning of the nav set here will change as Decanter overlays the mobile nav on top of other content and our designs have the mobile nav inline with the content.
```
can you break this comment into two lines so it is a more readable line length? thanks!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:
```diff
@@ -10,37 +9,46 @@
   input {
     @include font-smoothing;
-    @include padding(0.6rem 2rem);
+    padding: hb-calculate-rems(12px) hb-calculate-rems(36px) hb-calculate-rems(10px) hb-calculate-rems(14px);
```
👍

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:
```diff
@@ -10,37 +9,46 @@
 
     display: inline-block;
-    height: 4rem;
+    height: hb-calculate-rems(38px);
```
:good catch!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:
```diff
@@ -10,37 +9,46 @@
       opacity: 1; // For Firefox
+
+      @include hb-colorful {
+        color: darken($su-color-driftwood, 35%);
```
for the driftwood color, you dont have to use the hb-colorful theme mixin if you dont want!

heatherdesigns replied:
Good to know! So driftwood is going to be consistent throughout all the themes then?

silverli replied:
yes! it's a part of the stanford identity colors

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr499-nuinpeaifcojjeikvruwxzkyejynzz52.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e58067f8c1e65164ef749e8

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:
```diff
@@ -10,37 +9,46 @@
+    right: hb-calculate-rems(14px);
+    width: hb-calculate-rems(18px);
+    height: hb-calculate-rems(15px);
+    transform: scaleX(-1);
```
just curious, what gets scaled?

heatherdesigns replied:
Nope, it just flips it. It's a hack I learned on a previous project. 😄

silverli replied:
💥

silverli reviewed (changes requested):
looks pretty good! just a couple small requests :)

Commit:
feat: reposition nav dropdowns to allow the search input to be visible at all times

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr499-nuinpeaifcojjeikvruwxzkyejynzz52.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e58067f8c1e65164ef749e8

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_site-search.scss:
```diff
@@ -10,37 +9,42 @@
 
     &::placeholder {
-      color: $su-color-black;
+      color: darken($su-color-driftwood, 35%);
```
👍

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/settings/_variables.forms.scss:
```diff
@@ -3,6 +3,7 @@ $hb-input-min-height: 38px;
 
 $hb-input-border-radius: 4px;
 $hb-input-border: 1px solid $su-color-driftwood;
+$hb-input-max-width: 265px;
```
👍

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/settings/_variables.zindex.scss:
```diff
@@ -19,4 +19,5 @@ $hb-z-index-12: 1200;
 // $hb-z-index-modal: $hb-z-index-1;
 $hb-z-index-main-menu: $hb-z-index-1;
 $hb-z-index-lockup: $hb-z-index-5; // keeps the lockup from laying on top of the admin menu
+$hb-z-index-search: $hb-z-index-5; // keeps the search input from being hidden by the navigation
```
nice!

silverli reviewed (commented):
some really weird stuff happening with ie 11. let me know if you can replicate. we can always make a bug card if it feels out of scope


<img width="312" alt="Screen Shot 2020-02-28 at 11 43 58 AM" src="https://user-images.githubusercontent.com/12848168/75567794-1a9a6200-5a20-11ea-8246-6642797c0673.png">
<img width="865" alt="windows-10-ie-11" src="https://user-images.githubusercontent.com/12848168/75567795-1a9a6200-5a20-11ea-8cfc-b379b71bb7f2.png">
